crates/db_perf/src/main.rs

```diff
@@ -0,0 +1,174 @@
+use clap::Parser;
+use diesel::{
+  dsl::{self, sql},
+  sql_types,
+  ExpressionMethods,
+  IntoSql,
+};
+use diesel_async::{RunQueryDsl, SimpleAsyncConnection};
+use lemmy_db_schema::{
+  schema::post,
+  source::{
+    community::{Community, CommunityInsertForm},
+    instance::Instance,
+    person::{Person, PersonInsertForm},
+  },
+  traits::Crud,
+  utils::{
+    build_db_pool,
+    get_conn,
+    now,
+    series::{self, ValuesFromSeries},
+  },
+  SortType,
+};
+use lemmy_db_views::{post_view::PostQuery, structs::PaginationCursor};
+use lemmy_utils::error::LemmyResult;
+use std::num::NonZeroU32;
+
+#[derive(Parser, Debug)]
+struct CmdArgs {
+  #[arg(long, default_value_t = 3.try_into().unwrap())]
+  communities: NonZeroU32,
+  #[arg(long, default_value_t = 3.try_into().unwrap())]
+  people: NonZeroU32,
+  #[arg(long, default_value_t = 100000.try_into().unwrap())]
+  posts: NonZeroU32,
+  #[arg(long, default_value_t = 0)]
+  read_post_pages: u32,
+  #[arg(long)]
+  explain_insertions: bool,
+}
+
+#[tokio::main]
+async fn main() -> LemmyResult<()> {
+  if let Err(err) = try_main().await {
+    println!("😂 Error: {err:?}");
+  }
+  if let Ok(path) = std::env::var("PGDATA") {
+    println!("🪵 query plans and error details written in {path}/log");
+  }
+
+  Ok(())
+}
```

Comment:
I think this would make more sense as an [integration test](https://doc.rust-lang.org/rust-by-example/testing/integration_testing.html) in db_schema instead of a separate crate.

Reply:
It should be run separately because it is only useful for benchmarking, it's redundant for testing, it takes a while to run, and it should be configurable.

Also, this change is actually part of #4285. Most things in this PR other than the added replaceable-schema.sql file and related stuff are from a temporary merge that I did so I don't have to change the migration later when the other PR is done.